thrust/mr/new.h

```diff
@@ -43,6 +43,10 @@ class new_delete_resource THRUST_FINAL : public memory_resource<>
 #if __cplusplus >= 201703L
         return ::operator new(bytes, std::align_val_t(alignment));
 #else
+        // don't allocate anything if the caller requested zero bytes
+        if(0 == bytes){
+           return nullptr;
```

Comment:
I will need to consult someone who understands the C++ library spec better as to whether we are allowed to do this or not. I'm not entirely sure if we are.